File: src/main/java/com/github/graqr/threshr/model/redsky/products/pdp/client/Category.java
package com.github.graqr.threshr.model.redsky.products.pdp.client;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.serde.annotation.Serdeable;

@Serdeable
public record Category(
        @JsonProperty("parent_category_id")
        String parentCategoryId,
        String name) {
}

File: src/main/kotlin/com/graqr/threshr/model/redsky/product/Category.kt
package com.graqr.threshr.model.redsky.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class Category(
    @param:JsonProperty("parent_category_id") val parentCategoryId: String, val name: String
) {}
